extensions/sql-migration/src/dialog/feedbackDialog.ts

```diff
@@ -0,0 +1,196 @@
+								CSSStyles: {
+									'margin': '0 0 0 10px',
+								},
+							})
+							.component(),
+					])
+					.withLayout({
+						width: '100%',
+						alignContent: 'flex-start',
+						flexFlow: 'row',
+					})
+					.component();
+
+				this._buttonGroup = view.modelBuilder
+					.flexContainer()
+					.withItems([
+						this._createFeedbackButton(view, 0, loc.FEEDBACK_DIALOG_RATING_1),
+						this._createFeedbackButton(view, 1, loc.FEEDBACK_DIALOG_RATING_2),
+						this._createFeedbackButton(view, 2, loc.FEEDBACK_DIALOG_RATING_3),
+						this._createFeedbackButton(view, 3, loc.FEEDBACK_DIALOG_RATING_4),
+						this._createFeedbackButton(view, 4, loc.FEEDBACK_DIALOG_RATING_5),
+					])
+					.withLayout({
+						alignContent: 'flex-start',
+						flexFlow: 'row',
+					})
+					.withProperties<azdata.ComponentProperties>({
+						display: 'inline-flex',
+						ariaLabel: loc.FEEDBACK_DIALOG_HEADING,
+						CSSStyles: {
+							'margin-left': '40px'
+						}
+					})
+					.component();
+
+				const feedbackInputBox = view.modelBuilder
+					.inputBox()
+					.withProperties<azdata.InputBoxProperties>({
+						rows: 3,
+						inputType: 'text',
+						multiline: true,
+						placeHolder: loc.FEEDBACK_DIALOG_PLACEHOLDER,
+						CSSStyles: {
+							'white-space': 'normal!important',
+						},
+					})
+					.component();
+
+				feedbackInputBox.onTextChanged(
+					value => this._feedbackText = value);
+
+				const privacyLink = view.modelBuilder
+					.hyperlink()
+					.withProperties<azdata.HyperlinkComponentProperties>({
+						label: loc.FEEDBACK_DIALOG_PRIVACY_LINK,
+						url: 'https://privacy.microsoft.com/privacystatement',
+						showLinkIcon: true
+					})
+					.component();
+
+				const formModel = view.modelBuilder
+					.formContainer()
+					.withFormItems([{
+						components: [
+							{
+								component: headingGroup,
+							},
+							{
+								component: this._buttonGroup,
+							},
+							{
+								component: feedbackInputBox,
+							},
+							{
+								component: privacyLink,
+							}
+						],
+						title: ''
+					}])
+					.withLayout({ width: '100%' })
+					.component();
+
+				await view.initializeModel(formModel);
+				await this._buttonGroup.items[0].focus();
+			});
+
+			this._dialog.okButton.label = loc.FEEDBACK_DIALOG_SUBMIT_BUTTON;
+			this._dialog.okButton.onClick(async () => await this._execute());
+
+			this._dialog.cancelButton.label = loc.FEEDBACK_DIALOG_CANCEL_BUTTON;
+			this._dialog.cancelButton.onClick(() => this._cancel());
+
+			azdata.window.openDialog(this._dialog);
+		}
+	}
+
+	private async _execute() {
+		sendSqlMigrationActionEvent(
+			TelemetryViews.SqlMigrationFeedbackDialog,
+			TelemetryActions.SendFeedback,
+			{
+				'FeedbackRating': this._feedbackRating?.toString() || '',
+				'FeedbackMessage': this._feedbackText?.substr(0, 500) || '',
+			});
+
+		await vscode.window.showInformationMessage(loc.FEEDBACK_DIALOG_SENT_MESSAGE);
```

Comment:
IIRC, Charles told me that the raw telemetry data will only be kept for a limited time, do we have a pipeline to save the feedback to a permanent storage? why not save the feedback as a GitHub issue instead?

Reply:
we still want to go forward with a telemetry event, but pausing this pr until we do the privacy review for the extension.